### PR TITLE
Add MACE-POLAR-1 pretrained models

### LINKED ISSUE #143: PolarMACE (MACE-Polar-1) support: missing keys in _computeMACE inputDict
The current _computeMACE function in macepotential.py does not support PolarMACE models (e.g. MACE-Polar-1), which were released in February 2026. PolarMACE requires several additional keys in the input dictionary that are not currently passed by openmmml.

When attempting to run a simulation with MACE-Polar-1 through openmmml, the following errors are encountered sequentially:
openmm.OpenMMException: 'fermi_level'
openmm.OpenMMException: 'rcell'
openmm.OpenMMException: 'volume'
And an index out of bounds error related to missing head key.
Root Cause
PolarMACE's forward method requires the following additional keys in the input dictionary compared to regular MACE:

fermi_level — shape (num_graphs,), initialize to zeros
external_field — shape (num_graphs, 3), initialize to zeros
rcell — shape (num_graphs, 3, 3), reciprocal cell matrix (inverse of cell)
volume — shape (num_graphs,), volume of simulation box (determinant of cell)
head — shape (num_atoms,), per-atom head index (zeros for single-head models)

The following patch to _computeMACE resolves the issue:
pythoninputDict = {
    "ptr": ptr,
    "node_attrs": node_attrs,
    "batch": batch,
    "pbc": pbc,
    "positions": torch.tensor(positions, dtype=dtype, device=ptr.device),
    "edge_index": torch.tensor(edgeIndex, dtype=torch.int64, device=ptr.device),
    "shifts": torch.tensor(shifts, dtype=dtype, device=ptr.device),
    "cell": torch.tensor(cell, dtype=dtype, device=ptr.device),
    "total_charge": charge,
    "total_spin": multiplicity,
    # PolarMACE additional keys
    "fermi_level": torch.zeros(1, dtype=dtype, device=ptr.device),
    "external_field": torch.zeros((1, 3), dtype=dtype, device=ptr.device),
    "rcell": torch.linalg.inv(torch.tensor(cell, dtype=dtype, device=ptr.device)).unsqueeze(0),
    "volume": torch.tensor([abs(float(torch.linalg.det(torch.tensor(cell, dtype=dtype))))], dtype=dtype, device=ptr.device),
    "head": torch.zeros(len(positions), dtype=torch.long, device=ptr.device),
}

Suggested fix:
Add a check for PolarMACE model type in addForces and pass the additional required keys when detected:
pythonfrom mace.modules.extensions import PolarMACE as PolarMACEModel
is_polar = isinstance(model, PolarMACEModel)
Then conditionally add the extra keys to inputDict when is_polar is True.
Environment

openmmml version: latest main branch
mace version: latest main branch
MACE-Polar-1-S model from https://github.com/ACEsuit/mace-foundations/releases
OpenMM 8.5.1

### PR DESCRIPTION
**Pull request by epretti**
Adds MACE-POLAR-1 pretrained MACE models and fixes #143.

Like MACELES, requires an extra dependency to be installed from a repository (`git+https://github.com/WillBaldwin0/graph_electrostatics.git@v0.4.0`) as [documented](https://mace-docs.readthedocs.io/en/latest/guide/polar_mace.html#installation); this is a versioned release though.

### CODE REVIEW

File: openmmml/models/macepotential.py [hunk 1]
```diff
@@ -259,6 +264,12 @@ def _computeMACE(state, model, ptr, node_attrs, batch, pbc, returnEnergyType, ch
     dtype = node_attrs.dtype
     cutoff = float(model.r_max.detach())
     edgeIndex, shifts, _, _ = get_neighborhood(positions, cutoff, [periodic, periodic, periodic], cell)
+    cellTensor = torch.tensor(cell, dtype=dtype, device=ptr.device)
+    volume = torch.linalg.det(cellTensor)
+    if torch.abs(volume) > 0:
+        rcell = 2 * torch.pi * torch.linalg.inv(cellTensor.mT)
+    else:
+        rcell = torch.zeros((3, 3), dtype=dtype, device=ptr.device)
```
Comment (peastman) [inline openmmml/models/macepotential.py]: This is more complicated than it needs to be.  Because of the [requirements](https://docs.openmm.org/latest/userguide/theory/05_other_features.html#periodic-boundary-conditions) we have for box vectors, the box volume is just the product of the diagonal elements and is guaranteed to be positive.

File: test/data/mace_energies.py [hunk 2]
```diff
@@ -22,6 +22,19 @@
 results['mace-omat-0-medium'] = atoms.get_potential_energy()
 atoms.calc = mace_omol('extra_large')
 results['mace-omol-0-extra-large'] = atoms.get_potential_energy()
+try:
+    import graph_longrange
+    has_graph_longrange = True
+except ImportError:
+    print('No MACE-POLAR reference energies will be generated; you must first install graph_longrange from https://github.com/WillBaldwin0/graph_electrostatics')
+    has_graph_longrange = False
+if has_graph_longrange:
```
Comment (peastman) [inline test/data/mace_energies.py]: You could get rid of the `has_graph_longrange` variable and just put the calculations inside the `try` block.

File: openmmml/models/macepotential.py (hunk 1 above)
Comment (epretti) [reply openmmml/models/macepotential.py]: This was copied from their implementation which was copied from inside the MACE library.  They probably can't make the same guarantees we can about the box vectors.  I'll update it.

**Comment by epretti** [comment]
I also want to ask them about whether the `external_field` and `fermi_level` are things that can/should be exposed for users to set.  What they sent me showed these set to default values, which is what I did here.  `external_field` is semi-documented in [one place](https://mace-docs.readthedocs.io/en/latest/guide/polar_mace.html#basic-inference) but `fermi_level` seems not to be.
